extensions/mssql/src/dashboard/bookWidget.ts

```diff
@@ -0,0 +1,115 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+import * as nls from 'vscode-nls';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import {
+	BookContributionProvider
+} from './bookExtensions';
+
+const localize = nls.loadMessageBundle();
+
+export function registerBooksWidget(bookContributionProvider: BookContributionProvider): void {
+	azdata.ui.registerModelViewProvider('books-widget', async (view) => {
+		const container = view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'column',
+			width: '100%',
+			height: '100%',
+			alignItems: 'left'
+		}).component();
+		const bookslocationContainer = view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'column',
+			width: '270px',
+			height: '100%',
+			alignItems: 'left',
+			position: 'absolute'
+		}).component();
+		const bookRow = view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'row'
+		}).component();
+		const tsgbooklink = view.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+			label: localize('troubleshootingBooks', 'Troubleshooting Book'),
+			title: localize('troubleshootingBooksTitle', 'Troubleshooting Book'),
+		}).component();
+		tsgbooklink.onDidClick(() => {
+			PromptForFolder(bookContributionProvider);
+		});
+
+		bookRow.addItem(tsgbooklink, {
+			CSSStyles: {
+				'width': '100%',
+				'color': '#0078d4',
+				'text-decoration': 'underline',
+				'padding-top': '10px',
+				'text-align': 'left'
+			}
+		});
+		container.addItem(bookRow, {
+			CSSStyles: {
+				'padding-left': '10px',
+				'border-top': 'solid 1px #ccc',
+				'box-sizing': 'border-box',
+				'user-select': 'text'
+			}
+		});
+		bookslocationContainer.addItem(container, {
+			CSSStyles: {
+				'padding-top': '25px',
+				'padding-left': '5px'
+			}
+		});
+		await view.initializeModel(bookslocationContainer);
+	});
+}
+
+async function PromptForFolder(bookContributionProvider: BookContributionProvider) {
+	let filter = {
+		'All files': ['*']
+	};
+	let uris = await vscode.window.showOpenDialog({
+		filters: filter,
+		canSelectFiles: false,
+		canSelectMany: false,
+		canSelectFolders: true,
+		openLabel: 'Pick Folder'
+	});
+	if (uris && uris.length > 0) {
+		let pickedFolder = uris[0];
+		saveBooksToFolder(pickedFolder, bookContributionProvider);
```

Comment:
You need to await this before prompting to reload

Reply:
Otherwise you won't catch errors, and we might reload before copy is complete

Reply:
added await to the call above! Thanks for catching this.